### PR TITLE
CORE-63: Add code to load extra fields when submitting

### PR DESCRIPTION
**Pull request by ianmcorvidae**
This pulls the `extra_requirements` column from `apps_htcondor_extra` and adds it to the submission.

Still needs the downstream jex-adapter etc. work, of course, but hopefully if I did anything too blatantly dumb it can get caught here :laughing:

### CODE REVIEW

File: src/apps/service/apps/de/jobs/common.clj [hunk 1]
```diff
@@ -114,6 +114,16 @@
        (take-while (comp nil? :external_app_id))
        (reduce #(.buildStep request-builder %1 %2) [])))
 
+(defn load-htcondor-extra-requirements
+  [app-id]
+  (select :apps_htcondor_extra
+          (fields [:extra_requirements])
+          (where {:app_id app-id})))
```
Comment (ianmcorvidae) [inline src/apps/service/apps/de/jobs/common.clj]: ... I've already realized I probably need to add `(first (first ))` or something here

**Comment by ianmcorvidae** [comment]
Cool, thanks for the review! I'll probably merge this after I finish the other work to make these fields... actually at all useful, lol

File: src/apps/service/apps/de/jobs/common.clj [hunk 2]
```diff
@@ -114,6 +114,17 @@
        (take-while (comp nil? :external_app_id))
        (reduce #(.buildStep request-builder %1 %2) [])))
 
+(defn load-htcondor-extra-requirements
```
Comment (psarando) [inline src/apps/service/apps/de/jobs/common.clj]: I'd make this function private (`defn-`), just to make it obvious it's not in use outside this namespace.

Comment (ianmcorvidae) [reply src/apps/service/apps/de/jobs/common.clj]: Ah, yeah, good idea.

**Pull request merged**